test/metrics/prom-metrics-linter/custom_linter_rules.go

```diff
@@ -56,10 +56,140 @@ func CustomLinterRules(problems []promlint.Problem, mf *dto.MetricFamily, operat
 		}
 	}
 
-	// Sort the problems by metric name
-	sort.Slice(newProblems, func(i, j int) bool {
-		return newProblems[i].Metric < newProblems[j].Metric
-	})
-
 	return newProblems
 }
+
+func CustomRecordingRuleValidation(rr recordingRule) []promlint.Problem {
+	var problems []promlint.Problem
+
+	name := rr.Record
+	if name == "" {
+		return problems
+	}
+
+	// Require name structure: level:metric:operations
```

Comment:
it's hard to enforce the metric segment in the name, since labels values can follow similar strings (*_*_..) so it's hard to identify the metrics name in the promql, also the expr might include many metrics with long names which is impossible to enforce using all the metrics in the name, so we might want to add to the error messages a link to best practices: https://prometheus.io/docs/practices/rules/